src/claim/Claim.ts

```diff
@@ -13,35 +13,68 @@
 /**
  * Dummy comment needed for correct doc display, do not remove
  */
-import CType from '../ctype/CType'
+import ICType from '../ctype/CType'
 import { verifyClaimStructure } from '../ctype/CTypeUtils'
-import Identity from '../identity/Identity'
 import IClaim from '../types/Claim'
+import IPublicIdentity from '../types/PublicIdentity'
 
-function verifyClaim(claimContents: object, cType: CType): boolean {
-  return verifyClaimStructure(claimContents, cType.schema)
+function verifyClaim(
+  claimContents: object,
+  cTypeSchema: ICType['schema']
+): boolean {
+  return verifyClaimStructure(claimContents, cTypeSchema)
 }
 
 export default class Claim implements IClaim {
-  public static fromObject(obj: IClaim): Claim {
-    const newClaim = Object.create(Claim.prototype)
-    return Object.assign(newClaim, obj)
+  public static fromClaim(claimInput: IClaim): Claim {
+    if (claimInput.cTypeSchema) {
+      if (!verifyClaim(claimInput.contents, claimInput.cTypeSchema)) {
+        throw Error('Claim not valid')
+      }
+    }
+    return new Claim(claimInput)
+  }
+
+  public static fromCTypeAndClaimContents(
+    ctypeInput: ICType,
+    claimContents: object,
+    claimOwner: IPublicIdentity['address']
+  ): Claim {
+    if (ctypeInput.schema !== undefined) {
+      if (!verifyClaim(claimContents, ctypeInput.schema)) {
+        throw Error('Claim not valid')
+      }
+    }
+    return new Claim({
+      cTypeHash: ctypeInput.hash,
+      cTypeSchema: null,
+      contents: claimContents,
+      owner: claimOwner,
+    })
   }
 
-  public cType: IClaim['cType']
+  public cTypeHash: IClaim['cTypeHash']
   public contents: IClaim['contents']
   public owner: IClaim['owner']
+  public cTypeSchema: ICType['schema'] | null
 
-  public constructor(
-    cType: CType,
-    contents: IClaim['contents'],
-    identity: Identity
-  ) {
-    if (!verifyClaim(contents, cType)) {
-      throw Error('Claim not valid')
+  public constructor(claimInput: IClaim) {
+    if (claimInput.cTypeSchema) {
```

Comment:
Can you make these checks consistent with what you have in the Attestation constructor?